operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java

```diff
@@ -92,7 +95,10 @@ public final void registerEventSource(String name, EventSource eventSource)
       if (e instanceof KubernetesClientException) {
```

Comment:
Should not we catch this exception and make these checks in the CustomResourceEventSource? 
(And just throw the exception further if this is not the exception we want to trasnlate)